src/workerd/io/actor-cache.c++

```diff
@@ -291,8 +301,21 @@ void ActorCache::touchEntry(Lock& lock, Entry& entry) {
 }
 
 void ActorCache::removeEntry(Lock& lock, Entry& entry) {
-  KJ_IASSERT(!entry.isDirty());
```

Comment:
Hm, I kind of appreciated that we had some protections against accidentally removing dirty entries (and ensuring their `countedDelete` is carried forward) rather than happily doing so silently here.